framework/include/cppmicroservices/BundleInitialization.h

```diff
@@ -20,22 +20,34 @@
 
 =============================================================================*/
 
-#ifndef US_BUNDLE_NAME
-#    error Missing US_BUNDLE_NAME preprocessor define
-#endif
+#ifndef CPPMICROSERVICES_BUNDLEINITIALIZATION_H_TYPES
+#define CPPMICROSERVICES_BUNDLEINITIALIZATION_H_TYPES
 
-#ifndef CPPMICROSERVICES_BUNDLEINITIALIZATION_H
-#    define CPPMICROSERVICES_BUNDLEINITIALIZATION_H
+#include "cppmicroservices/GlobalConfig.h"
 
-#    include "cppmicroservices/GlobalConfig.h"
-
-#    include <atomic>
+#include <functional>
 
 namespace cppmicroservices
 {
+    class BundleContext;
     class BundleContextPrivate;
+    // THIS IS A TRANSFER OF OWNERSHIP. The pointed object will be deleted when the bundle is
```

Comment:
I find this comment to be confusing. This actually does nothing except declare a type which is a std::function, which can be coded to do anything. Whether or not it transfers ownership depends entirely on how a particular std::function of that type is coded.

Reply:
These essentially act as function signatures for the functions inside `CPPMICROSERVICES_INITIALIZE_BUNDLE`. We can't use actual function signatures, since the implementations are in a runtime-selected bundle and therefore can't be resolved at link time. I will add a brief note to explain this. Since interface-level doc generally goes with the function signature, I decided to put it here.